scripts/serie_update.py

```diff
@@ -0,0 +1,590 @@
+        self.zephyr_module_serie_path =\
+            self.zephyr_hal_stm32_path / "stm32cube" / self.stm32_seriexx
+        self.stm32cube_temp_serie = \
+            self.stm32cube_temp / "stm32cube" / self.stm32_seriexx
+        self.stm32cube_temp_serie.mkdir(parents=True)
+
+        self.readme_file_path = self.zephyr_module_serie_path / "README"
+        self.version_tag = []
+        self.current_version = ""
+        self.latest_version = ""
+        self.latest_commit = ""
+
+    ###################################
+    # Clone or fetch a stm32 serie repo
+    #
+    def clone_cube_repo(self):
+        os.chdir(self.stm32cube_repo_path)
+        if self.stm32cube_serie_path.exists():
+            print("fetching repo " + str(self.stm32cube_serie_path))
+            # if already exists, then just clean and fetch
+            os.chdir(self.stm32cube_serie_path)
+            os_cmd("git clean -fdx")
+            os_cmd("git fetch")
+            os_cmd("git reset --hard master")
+        else:
+            cmd = (
+                "git clone https://github.com/STMicroelectronics/STM32Cube"
+                + self.serie
+                + ".git"
+            )
+            os_cmd(cmd)
+            os.chdir(self.stm32cube_serie_path)
+
+        # get the latest version of cube,
+        # with the most recent one created being the last entry.
+        os_cmd("git checkout master")
+        self.version_tag = subprocess.check_output("git tag -l",
+                                                   shell=True).splitlines()
+        self.version_tag = [x.decode("utf-8") for x in self.version_tag]
+        # Set latest version
+        self.latest_version = self.version_tag[-1]
+
+    ####################################
+    # Retruns current zephyr hal version
+    #
+    def get_zephyr_current_version(self):
+        os.chdir(self.zephyr_module_serie_path)
+
+        with open(str(self.readme_file_path), "r") as f:
+            for line in f:
+                # pattern : "version " follow by optional "v",
+                # followed by x.y or x.y.z  x,y,z may represent several digits
+                # ex: 'version v1.8.9', 'version 10.20.25'
+                pattern = r".*version v?(\d+\.\d+\.?\d*).*$"
+                if re.match(pattern, line):
+                    previous_version = re.sub(pattern, r"\1", line).rstrip('\n')
+                    break
+
+        # Match previous version and list of existing tags
+        # which could be vx.y or x.y
+        pos_version = \
+            [i for i, a in enumerate(self.version_tag) if previous_version in a]
+        if pos_version:
+            # return previous zephyr version
+            return self.version_tag[pos_version[0]]
+        else:
+            print("Error: cannot find version " + previous_version
+                  + " in STM32Cube_repo")
+            self.cleanup(self.stm32_serie, self.stm32cube_repo_path)
+            sys.exit()
```

Comment:
Better raise errors on classes, so that error handling is deferred to the final user, i.e. the actual CLI tool.

Reply:
Done